constellation_whitepaper_v0.1.tex

```diff
@@ -0,0 +1,70 @@
+
+Gossip protocols allow for large networks to communicate total network state. Each member of the network keeps track of neighbors and when it receives new messages, it propagates that message to all of its neighbors. This has some interesting mathematical properties, but in our case, it allows for a large pool of connected devices to share the state of their resources.
+
+Horizontal scalability inherently means splitting one program into multiple asynchronous operations. One approach to blockchain consensus that allows this is a TrustChain, namely ExtendedTrustChain. In ExtendedTrustChain, there are multiple node types, each with their own responsibility that asynchronously govern different aspects of the protocol. Specifically, there are basic nodes that send transactions and host individual chains (each individual's chain forms DAG's with others when a transaction is made and is hashed into linear blocks). There are node processes that perform consensus on checkpoints, collections of transactions that will be hashed and become the next block in the chain. Finally, there are validator processes, who host the current chain state and seed chain state/history to the node. Each node is associated with an account and a meme (for reputation.) This architecture will allow for personal devices to run full nodes and perform fast transactions.
+
+The formulation of distributed applications as MapReduce operations on smart contracts has been proposed by the Plasma upgrade to Ethereum. However with its application to the EVM, it still suffers the same pitfalls of synchronous architecture. The Solidity programming language was built to address the risks of non-determinism in smart contract development. The need for deterministic instruction stems from the need to calculate the cost of running a Smart Contract and prevent against DDOS attacks. However, if compiled byte code could be notarized on chain and verified in a JVM, then deterministic compilation becomes a looser requirement. Reduction in gas consumption, circumvention of gas limits and smart contracts with the complexity of microservices are now a possibility.
+
+It is logical to design Dapps as compositions of successful building blocks. On chain interoperability allows for blockchain companies to provide services and use others. A truly decentralized ecosystem cannot exist with monolithic smart contracts. A successful ecosystem tends towards diversity. Successful ecological actors siphon energy from more antiquated species and their traits die out. 
+
+The use of the JVM as a conduit for distributed applications is an industry standard in the Big Data community. Consumer applications are deployed as a collection of microservices on autoscaling groups which rely on easy provisioning, like the JVM provides. Thousand node clusters can be provisioned and autoscaled on AWS virtual instances and run distributed applications on the JVM. The same can be of smart contract microservices. 
```

Comment:
There is a potential to make a transcompiler from JVM to EVM, that woukd allow java code, and therefore scala code to be able to run on the ethereum network.

Reply:
Yeah, although I think it might be easier to use oracles in ethereum to communicate with microservices on constellation. Thinking of making a complimentary interface in solidity. Not sure if you could mirror the same service with a port of JVM opcodes to EVM. What do you think?